File: app/src/main/java/com/hmmelton/kibbble/adapters/AdoptablePetsAdapter.java
package com.hmmelton.kibbble.adapters;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.hmmelton.kibbble.R;
import com.hmmelton.kibbble.models.Pet;

import java.util.List;

import butterknife.ButterKnife;

/**
 * Created by harrison on 4/16/17.
 * This is an adapter class for a list containing adoptable pets.
 */

public class AdoptablePetsAdapter extends RecyclerView.Adapter<AdoptablePetsAdapter.ViewHolder> {

    private List<Pet> mPets; // Adoptable pets to display

    /**
     * Constructor
     * @param pets list of adoptable pets to display
     */
    public AdoptablePetsAdapter(List<Pet> pets) {
        mPets = pets;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        // Inflate layout resource
        View rootView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.card_view, parent, false);

        // Pass layout to view holder
        return new ViewHolder(rootView);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {

    }

    @Override
    public int getItemCount() {
        // TODO: change this to mPets.size()
        return 4;
    }

    /**
     * This method adds a Pet object to the end of the adapter.
     * @param pet Pet to be added
     */
    public void addView(Pet pet) {
        mPets.add(pet);
    }

    /**
     * View holder class for enclosing RecyclerView
     */
    class ViewHolder extends RecyclerView.ViewHolder {

        public ViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/com/hmmelton/kibbble/adapters/AdoptablePetsAdapter.kt
package com.hmmelton.kibbble.adapters

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import butterknife.BindView
import butterknife.ButterKnife
import com.bumptech.glide.Glide
import com.hmmelton.kibbble.R
import com.hmmelton.kibbble.models.Pet

/**
 * Created by harrisonmelton on 4/30/17.
 * This is an adapter class for a list containing adoptable pets.
 */
class AdoptablePetsAdapter(private val mPets: MutableList<Pet>) :
        RecyclerView.Adapter<AdoptablePetsAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        // Inflate layout resource
        val rootView = LayoutInflater.from(parent.context)
                .inflate(R.layout.card_view, parent, false)

        // Pass layout to view holder
        return ViewHolder(rootView)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        // Get pet at given position
        val pet = mPets[position]
        // Set image
        val imageUrlString = pet.images?.get(0)
        if (imageUrlString != null) {
            Glide.with(holder.profileImageView?.context)
                    .load(imageUrlString)
                    .into(holder.profileImageView)
        }
        // Set name
        holder.nameAgeTextView?.text = pet.name + ", " + pet.age
        // Set gender
        holder.genderTextView?.text = pet.gender
    }

    override fun getItemCount(): Int {
        return mPets.size
    }

    /**
     * This method adds a Pet object to the end of the adapter.
     * @param pet Pet to be added
     */
    fun addView(pet: Pet) {
        mPets.add(pet)
        notifyItemInserted(mPets.size - 1)
    }

    /**
     * View holder class for enclosing RecyclerView
     */
    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        @BindView(R.id.iv_card_item)
        var profileImageView: ImageView? = null
        @BindView(R.id.tv_name_age_card_item)
        var nameAgeTextView: TextView? = null
        @BindView(R.id.tv_gender_card_item)
        var genderTextView: TextView? = null

        init {
            ButterKnife.bind(this, itemView)
        }
    }
}
